zyte_common_items/pipelines.py

```diff
@@ -53,3 +57,49 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropItem(Exception):
+    """Drop item from the item pipeline."""
+
+    pass
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        o = cls(crawler)
+        return o
+
+    def init_thresholds(self, spider):
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = item if isinstance(item, str) else global_object_name(item)
+            self.thresholds[item_type] = threshold
+
+    def get_threshold(self, item, spider):
+        return self.thresholds.get(
+            global_object_name(item.__class__), DEFAULT_ITEM_PROBABILITY_THRESHOLD
+        )
+
+    async def process_item(self, item, spider):
+        threshold = self.get_threshold(item, spider)
+
+        self.stats.inc_value("item/crawl/total", spider=spider)
+        item_proba = item.get_probability()
+        if item_proba is None or item_proba >= threshold:
+            self.stats.inc_value("item/crawl/extracted_with_high_proba", spider=spider)
```

Comment:
Have you thought of any alternative names for the stats? I wonder if something like `"item_probability_threshold/dropped"/"item_probability_threshold/kept"` may be best.

I also wonder if we really need `"item_probability_threshold/kept"`, as I expect its value to match `item_scraped_count`.

```suggestion
            self.stats.inc_value(, spider=spider)
```

Reply:
Hmm… Looking for existing stats about dropping stuff to keep the naming consistent, I found we already have a stat for this in the e-commerce spider: https://github.com/zytedata/zyte-spider-templates/blob/5f4c2c667d0a33029ed3ebcee1692d9ce61d98e4/zyte_spider_templates/spiders/ecommerce.py#L171

Maybe we could keep the same name here, and eventually modify the ecommerce spider to use this middleware instead of its own dropping logic, with no change to the stat name: `drop_item/product/low_probability`. But then we would need to have some mapping from known item classes to their Zyte API parameter name (i.e. `zyte_common_items.Product` → `product`). Maybe we can simply do `name = item.__class__.__name__; name[0] = name[0].lower()`.

Reply:
Haven't thought much about the names here, just left that I used in Articles. Using both stats (kept/dropped) I considered quite useful as they are close to each other in stats, it's clear where these metrics come from and easy to compare. You are right, in general case kept==scrapped, but if suddenly that are different - it can be helpful to start searching a bug.
Anyway, might be at this moment to have compatibility with e-commerce, we can keep only drop_item/{name}/low_probability.

Reply:
@Gallaecio based on discussion [here](https://github.com/zytedata/zyte-common-items/pull/91#pullrequestreview-2004355909) and  if we are thinking about not using names like in e-commerce stats - what do you think about it?:
1. using this approach 
```
  self.stats.inc_value(
          f"probability_drop_item_pipeline/input_items/{self.get_item_name(item)}", spider=spider
      )
  item_proba = item.get_probability()
  if item_proba is None or item_proba >= threshold:
      self.stats.inc_value(
          f"probability_drop_item_pipeline/kept_items/{self.get_item_name(item)}", spider=spider
      )
      return item
  self.stats.inc_value(
          f"probability_drop_item_pipeline/dropped_items/{self.get_item_name(item)}", spider=spider
      )
```
2. Rename the pipeline to ProbabilityDropItemPipeline
cc @BurnzZ @wRAR @kmike